repair/repair.cc

```diff
@@ -1030,9 +1033,109 @@ future<> repair::shard_repair_task_impl::release_resources() noexcept {
     neighbors = {};
     dropped_tables = {};
     nodes_down = {};
+    _update_erm_callback = {};
     return make_ready_future();
 }
 
+static future<bool> ranges_intersect(const dht::token_range_vector& ranges, const dht::token_range& range) {
+    for (const auto& r : ranges) {
+        if (range.intersection(r, dht::token_comparator()).has_value()) {
+            co_return true;
+        }
+        co_await coroutine::maybe_yield();
+    }
+    co_return false;
+}
+
+static future<bool> compare_tablets(const locator::effective_replication_map& old_erm, const locator::effective_replication_map& new_erm, table_id table_id, const std::unordered_set<locator::tablet_id>& involved_tablets) {
+    auto& old_tablet_map = old_erm.get_token_metadata().tablets().get_tablet_map(table_id);
+    auto& new_tablet_map = new_erm.get_token_metadata().tablets().get_tablet_map(table_id);
+    if (old_tablet_map.tablet_count() != new_tablet_map.tablet_count()) {
+        co_return false;
+    }
+    auto tablet_changed = [&] (locator::tablet_id tid) {
+        try {
+            auto old_transition = old_tablet_map.get_tablet_transition_info(tid);
+            auto new_transition = new_tablet_map.get_tablet_transition_info(tid);
+            return (old_transition && new_transition ? *old_transition != *new_transition : old_transition || new_transition) ||
```

Comment:
How can we end up with old_transition is null while repair is running? Repair must be run in the repair transition. How can we transit to a new  transition while the old transition is not finished yet.

Reply:
Outdated